### PR TITLE
Rename ConfigureResource -> ConfigureResourceBuilder

### PR DESCRIPTION
**Pull request by alanwest**
Staging this up based on open question here https://github.com/open-telemetry/opentelemetry-dotnet/pull/3307#discussion_r907543231.

### CODE REVIEW

File: src/OpenTelemetry/CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,11 @@
 * Handle possible exception when initializing the default service name.
   ([#3405](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3405))
 
+* Add `ConfigureResourceBuilder` which can replace SetResourceBuilder more
+  succinctly in most cases and has greater flexibility (applies to
+  TracerProvicerBuilder, MeterProviderBuilder, OpenTelemetryLoggingOptions).
```
Comment (cijothomas) [inline src/OpenTelemetry/CHANGELOG.md:30]:
```suggestion
  TracerProviderBuilder, MeterProviderBuilder, OpenTelemetryLoggingOptions).
```

**Comment by Rs4178** [comment]
> Staging this up based on open question here https://github.com/open-telemetry/opentelemetry-dotnet/pull/3307#discussion_r907543231.

**Review by cijothomas — APPROVED** [review]
I like this name better. Lets wait couple days to give more people to share feedbacks before merge.

File: examples/Console/TestConsoleExporter.cs [hunk 2]
```diff
@@ -39,7 +39,7 @@ private static object RunWithActivitySource()
             // and use Console exporter.
             using var tracerProvider = Sdk.CreateTracerProviderBuilder()
                     .AddSource("Samples.SampleClient", "Samples.SampleServer")
-                    .ConfigureResource(res => res.AddService("console-test"))
+                    .ConfigureResourceBuilder(res => res.AddService("console-test"))
```
Comment (reyang) [inline examples/Console/TestConsoleExporter.cs:42]: It seems the current code are bit random on the local variable naming.
Maybe use `resourceBuilder` or `resBldr` consistently across the repo?

**Review by reyang — APPROVED** [review]

**Comment by Oberon00** [comment]
I liked the old/current name a bit more. Just as in ASP.NET one uses ConfigureServices, not ConfigureServiceCollection. I could argue, I don't want to configure the builder, but use the builder to configure the resource(s) that will be created. But this is not a strong opinion.

**Comment by alanwest** [comment]
> I liked the old/current name a bit more. Just as in ASP.NET one uses ConfigureServices, not ConfigureServiceCollection. Or this method here, another example where one uses a Builder to configure something but which doesn't have Builder in the name: https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.testing.webapplicationfactory-1.configurewebhost?view=aspnetcore-6.0 I could argue, I don't want to configure the builder, but use the builder to configure the resource(s) that will be created. But I do not have a very strong opinion.

I agree, these are good examples and a good argument for keeping the name as is: `ConfigureResource`.

**Comment by cijothomas** [comment]
> I liked the old/current name a bit more. Just as in ASP.NET one uses ConfigureServices, not ConfigureServiceCollection. Or this method here, another example where one uses a Builder to configure something but which doesn't have Builder in the name: https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.testing.webapplicationfactory-1.configurewebhost?view=aspnetcore-6.0 I could argue, I don't want to configure the builder, but use the builder to configure the resource(s) that will be created. But I do not have a very strong opinion.

Thanks for explanation. It looks reasonable to me to keep existing name, and abort this PR.

**Pull request closed without merging**